Assets/Scripts/UI/TutorialPlayerThoughts.cs

```diff
@@ -0,0 +1,42 @@
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class TutorialPlayerThoughts : MonoBehaviour
+{
+    public TextMeshProUGUI playerThoughts;
+    private bool _bounce = false;
+    private int _num_bounces = 0;
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (_bounce)
+        {
+            if (_num_bounces > 6)
+            {
+                playerThoughts.text = "The power generator needs to be turned off.";
+            }
+            else if (_num_bounces > 3)
+            {
+                playerThoughts.text = "Maybe the control panel will be helpful.";
+            }
+            else
+            {
+                playerThoughts.text = "Ouch. The water is electrified by the power generator.";
+            }
+
+        }
```

Comment:
I think we can move this logic to under the `OnControllerColliderHit`. That way we would not be doing these checks every frame. So something like this maybe?

```C#
if (hit.gameObject.CompareTag("puddle"))
{
    _num_bounces += 1;
    
    if (_num_bounces > 6)
    { .. }
    etc...
}
  ```

Reply:
fixed, thanks for the catch :D